src/core/shape_inference/include/irdft_shape_inference.hpp

```diff
@@ -0,0 +1,140 @@
+        // [n_0, ..., n_{r - 1}].
+        if (axes_shape.rank().is_static() && axes_are_known) {
+            for (int64_t& axis : axes) {
+                if (axis < 0) {
+                    axis += input_rank - 1;
+                }
+            }
+
+            ov::AxisSet axes_set;
+            for (const auto& axis : axes) {
+                axes_set.insert(static_cast<size_t>(axis));
+            }
+
+            NODE_VALIDATION_CHECK(op, axes.size() == axes_set.size(), "FFT op axes must be unique.");
+
+            NODE_VALIDATION_CHECK(op,
+                                  std::find(axes.begin(), axes.end(), input_rank - 1) == axes.end(),
+                                  "IRDFT op axes cannot contain the last axis.");
+        }
+    }
+
+    NODE_VALIDATION_CHECK(op, axes_shape.rank().compatible(1), "IRDFT op axes input must be 1D tensor.");
+
+    if (input_shapes.size() == 3) {
+        const auto& signal_size_shape = input_shapes[2];
+        NODE_VALIDATION_CHECK(op,
+                              signal_size_shape.rank().compatible(1),
+                              "IRDFT op signal size input must be 1D tensor. Got signal: ",
+                              signal_size_shape);
+
+        if (axes_shape.is_static() && signal_size_shape.is_static()) {
+            NODE_VALIDATION_CHECK(op,
+                                  axes_shape[0].compatible(signal_size_shape[0]),
+                                  "Sizes of inputs 'axes' and 'signal_size' must be equal. Got "
+                                  "size of 'axes': ",
+                                  axes_shape[0],
+                                  "size of 'signal_size': ",
+                                  signal_size_shape[0]);
+        }
+    }
+
+    if (input_shape.rank().is_dynamic()) {
+        output_shape = ov::PartialShape();
+        return;
+    }
+
+    const auto input_rank = input_shape.size();
+
+    output_shape = input_shape;
+    output_shape.resize(input_rank - 1);
+
+    if (axes_shape.rank().is_dynamic() || !axes_are_known) {
+        for (int64_t i = 0; i < input_rank - 1; ++i) {
+            output_shape[i] = ov::Dimension::dynamic();
+        }
+        return;
+    }
+
+    const auto last_axis = axes.back();
+
+    if (input_shapes.size() == 2) {
+        output_shape[last_axis] = DimType(2) * (input_shape[last_axis] - DimType(1));
+        return;
+    }
+
+    const auto& signal_size_shape = input_shapes[2];
+    std::vector<int64_t> signal_size;
+    bool status_signal_size = get_data_as_int64<T>(2, op, signal_size, constant_data);
+
+    if (signal_size_shape.rank().is_dynamic() || !status_signal_size) {
+        output_shape[last_axis] = ov::Dimension::dynamic();
+        return;
+    }
+
+    size_t num_of_axes = axes.size();
+    for (size_t i = 0; i < num_of_axes; ++i) {
+        if (signal_size[i] != -1) {
+            output_shape[axes[i]] = DimType(signal_size[i]);
+        }
+    }
+    if (signal_size.back() == -1) {
```

Comment:
If signal_size.back() == -1, then last_axis  == -1. We will have index error in output_shape[last_axis].

Reply:
If `signal_size.back() == -1`, it doesn't mean that `last_axis == -1`, because `last_axis` is defined as `const auto last_axis = axes.back();`, not as `signal_size.back()`.